### ISSUE TITLE
Importing qstash broken in 2.7.0 due to import path

### ISSUE THREAD

**Issue by mozeryansky** [upstash/qstash-js#167]
The newly released 2.7.0 version does not work, use the nodejs example to reproduce. It does work in 2.6.5.

I believe is it due to: [Fix import paths #164](https://github.com/upstash/qstash-js/pull/164)

```
~/repos/tmp/qstash-js/examples/nodejs (main) $ node index.js 
node:internal/modules/cjs/loader:1148
  throw err;
  ^

Error: Cannot find module '@upstash/qstash'
Require stack:
- /Users/michael/repos/tmp/qstash-js/examples/nodejs/index.js
    at Module._resolveFilename (node:internal/modules/cjs/loader:1145:15)
    at Module._load (node:internal/modules/cjs/loader:986:27)
    at Module.require (node:internal/modules/cjs/loader:1233:19)
    at require (node:internal/modules/helpers:179:18)
    at Object.<anonymous> (/Users/michael/repos/tmp/qstash-js/examples/nodejs/index.js:1:30)
    at Module._compile (node:internal/modules/cjs/loader:1358:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1416:10)
    at Module.load (node:internal/modules/cjs/loader:1208:32)
    at Module._load (node:internal/modules/cjs/loader:1024:12)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:174:12) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [ '/Users/michael/repos/tmp/qstash-js/examples/nodejs/index.js' ]
}

Node.js v20.14.0
```

**Comment by ogzhanolguncu** [comment]
We dropped the support of cjs for qstash. Since we thought there were not enough cjs users. Nodejs is compatible with ESM can you try to use it like this?

**Comment by ogzhanolguncu** [comment]
Try to use of the examples with latest build. Right now I'm double checking everything.d

**Comment by mozeryansky** [comment]
1) I cloned the rep
2) `npm i` then `npm run build`
3) `cd examples/nodejs` then `pnpm i` (since there's a pnpm lock file, just to be sure)
4) `node index.js`

```
$ node index.js
(node:40842) Warning: To load an ES module, set "type": "module" in the package.json or use the .mjs extension.
(Use `node --trace-warnings ...` to show where the warning was created)
/Users/michael/repos/tmp/qstash-js/examples/nodejs/index.js:1
import { Client, Receiver } from "@upstash/qstash";
^^^^^^

SyntaxError: Cannot use import statement outside a module
    at wrapSafe (node:internal/modules/cjs/loader:1281:20)
    at Module._compile (node:internal/modules/cjs/loader:1321:27)
    at Module._extensions..js (node:internal/modules/cjs/loader:1416:10)
    at Module.load (node:internal/modules/cjs/loader:1208:32)
    at Module._load (node:internal/modules/cjs/loader:1024:12)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:174:12)
    at node:internal/main/run_main_module:28:49

Node.js v20.14.0
```

When I added `"type": "module"` to the package file I get this error:

```
$ node index.js
node:internal/modules/esm/resolve:215
  const resolvedOption = FSLegacyMainResolve(packageJsonUrlString, packageConfig.main, baseStringified);
                         ^

Error: Cannot find package '/Users/michael/repos/tmp/qstash-js/examples/nodejs/node_modules/@upstash/qstash/package.json' imported from /Users/michael/repos/tmp/qstash-js/examples/nodejs/index.js
    at legacyMainResolve (node:internal/modules/esm/resolve:215:26)
    at packageResolve (node:internal/modules/esm/resolve:841:14)
    at moduleResolve (node:internal/modules/esm/resolve:927:18)
    at defaultResolve (node:internal/modules/esm/resolve:1157:11)
    at ModuleLoader.defaultResolve (node:internal/modules/esm/loader:383:12)
    at ModuleLoader.resolve (node:internal/modules/esm/loader:352:25)
    at ModuleLoader.getModuleJob (node:internal/modules/esm/loader:227:38)
    at ModuleWrap.<anonymous> (node:internal/modules/esm/module_job:87:39)
    at link (node:internal/modules/esm/module_job:86:36) {
  code: 'ERR_MODULE_NOT_FOUND'
}

Node.js v20.14.0
```

**Comment by ogzhanolguncu** [comment]
I'll fix the nodejs example and convert it to esm.

**Comment by mozeryansky** [comment]
Just to add, with the latest version (2.7.0) I can confirm it works when specifying the full path to the mjs file:

```
import { Client, Receiver } from '../node_modules/@upstash/qstash/dist/base/index.mjs'
```

**Comment by ogzhanolguncu** [comment]
Exactly this. If you are using `js`, all you have to do this `@upstash/qstash/base/index.mjs` and add `type:"module"` to package.json. If you wanna fix the example go ahead or I'll.

**Comment by mozeryansky** [comment]
Go ahead, you could include it in your linked PR since it will close this ticket.

**Comment by mozeryansky** [comment]
I've pulled from main and verified everything, works great! Thanks!

**Comment by mozeryansky** [comment]
tl;dr: I noticed there are no files in `node_modules/@upstash/qstash`.

--

Now I get a different error when testing the package in my project, can you confirm?

I've installed from the latest commit using `npm i https://github.com/upstash/qstash-js.git\#f86ad8a`, and ran my script and get this error:

```
$ tsx scripts/qstash.ts                                  
node:internal/modules/esm/resolve:304
  return new ERR_PACKAGE_PATH_NOT_EXPORTED(
         ^

Error [ERR_PACKAGE_PATH_NOT_EXPORTED]: Package subpath './dist/base/index.mjs' is not defined by "exports" in /Users/michael/repos/test/node_modules/@upstash/qstash/package.json
    at exportsNotFound (node:internal/modules/esm/resolve:304:10)
    at packageExportsResolve (node:internal/modules/esm/resolve:651:9)
    at resolveExports (node:internal/modules/cjs/loader:592:36)
    at Module._findPath (node:internal/modules/cjs/loader:669:31)
    at Module._resolveFilename (node:internal/modules/cjs/loader:1131:27)
    at resolve (/usr/local/lib/node_modules/tsx/dist/register-DfubRCxM.cjs:1:3084)
    at resolveRequest (/usr/local/lib/node_modules/tsx/dist/register-DfubRCxM.cjs:1:2618)
    at /usr/local/lib/node_modules/tsx/dist/register-DfubRCxM.cjs:1:3400
    at m._resolveFilename (file:///usr/local/lib/node_modules/tsx/dist/register-CFO5XQXL.mjs:1:832)
    at Module._load (node:internal/modules/cjs/loader:986:27) {
  code: 'ERR_PACKAGE_PATH_NOT_EXPORTED'
}

Node.js v20.14.0
```

I'm importing using: `import { Client, Receiver } from '@upstash/qstash/dist/base/index.mjs'`.

-- 

Reverting shows this is a regression is still broken:

`npm i @upstash/qstash@v2.6.5` will work

**Comment by ytkimirti** [comment]
You could not have installed the project from github if we only used js. But since we are using typescript the code needs to be built and put into the `dist` folder. Thats why when you install from git, there are no files.

To test the latest version, you can clone the repo yourself, run `bun install` and `bun run build`.

Then you can install the package from the root folder like this
`npm i ../proj/upstash/qstash-js`

**Comment by mozeryansky** [comment]
@ytkimirti Ok, thanks. I did that to verify the file name mispelling, which I believe is the main cause of this bug (and #168). Could you verify the PR? https://github.com/upstash/qstash-js/pull/174
